android/expoview/src/main/java/host/exp/exponent/experience/ReactNativeActivity.java

```diff
@@ -691,26 +690,7 @@ public void onRequestPermissionsResult(final int requestCode, final String[] per
   @Override
   public int checkPermission(final String permission, final int pid, final int uid) {
     int globalResult = super.checkPermission(permission, pid, uid);
-
-    // only these permissions, which show a dialog to the user should be scoped.
-    boolean isDangerousPermission;
-    try {
-      PermissionInfo permissionInfo = getPackageManager().getPermissionInfo(permission, PackageManager.GET_META_DATA);
-      isDangerousPermission = (permissionInfo.protectionLevel & PermissionInfo.PROTECTION_DANGEROUS) != 0;
-    } catch (PackageManager.NameNotFoundException e) {
-      return PackageManager.PERMISSION_DENIED;
-    }
-
-    if (Constants.isStandaloneApp() || !isDangerousPermission) {
-      return globalResult;
-    }
-
-    if (globalResult == PackageManager.PERMISSION_GRANTED &&
-        mExpoKernelServiceRegistry.getPermissionsKernelService().hasGrantedPermissions(permission, mExperienceId)) {
-      return PackageManager.PERMISSION_GRANTED;
-    } else {
-      return PackageManager.PERMISSION_DENIED;
-    }
+    return mExpoKernelServiceRegistry.getPermissionsKernelService().getFinalPermissions(globalResult, getPackageManager(), permission, mExperienceId);
```

Comment:
```suggestion
    return mExpoKernelServiceRegistry.getPermissionsKernelService().getPermissions(globalResult, getPackageManager(), permission, mExperienceId);
```

Why `finalPermissions`? 🤔 IS there any hidden sense I'm not getting here?